elide-graphql/src/main/java/com/yahoo/elide/graphql/subscriptions/websocket/SubscriptionWebSocket.java

```diff
@@ -231,21 +258,52 @@ private SessionHandler findSession(Session wrappedSession) {
         return sessionHandler;
     }
 
+    @SuppressWarnings("unchecked")
     protected SessionHandler createSessionHandler(Session session) {
-        String apiVersion = session.getRequestParameterMap().getOrDefault("ApiVersion",
-                List.of(NO_VERSION)).get(0);
-
         User user = userFactory.create(session);
 
+        String path = session.getPathParameters().get("path");
+        if (path == null) {
+           path = "";
+        }
+
+        String baseUrl = getBaseUrl(session);
+        if (!path.isBlank() && baseUrl.endsWith(path)) {
+            baseUrl = baseUrl.substring(0, baseUrl.length() - path.length());
+        }
+
+        Map<String, List<String>> headers = new LinkedHashMap<>(session.getRequestParameterMap());
+        // Get the properties from the handshake set in SubscriptionWebSocketConfigurator
+        if (session.getUserProperties().get("headers") instanceof Map handshakeHeaders) {
+            headers.putAll(handshakeHeaders);
+        }
+
+        Route route = routeResolver.resolve(MediaType.APPLICATION_JSON, baseUrl, path, headers,
+                session.getRequestParameterMap());
+
+        String apiVersion = route.getApiVersion();
```

Comment:
The route resolver here that resolved the apiVersion from parameters looks at a different parameter name than the old code.  The old code looked for ApiVersion in the parameters and the new code looks for "v" in the ParameterRouteResolver.

Reply:
The `ApiVersion` that it looked for previously was more in line with what was looked for from the header. In the new code this is catered by initializing the header values from the request parameter map in the `createSessionHandler` using `Map<String, List<String>> headers = new LinkedHashMap<>(session.getRequestParameterMap());` in addition to the other values found as HTTP headers. So the `HeaderRouteResolver` should resolve the API Version if configured with `ApiVersion`.